src/sema/ast.rs

```diff
@@ -102,6 +102,11 @@ pub trait RetrieveType {
     fn ty(&self) -> Type;
 }
 
+/// This trait helps retrieving common members from Function, StructDecl and EnumDecl
+pub trait DeclarationInfo {
```

Comment:
This trait only provides `fn tags()`. 
 - I think should be called what they _are_, not what want them to be in the future (bc you can't predict how the code will change in future). How about calling it `Tags` and in future it can be renamed *when* it acquires more methods.
 - I don't know what this trait is supposed to add. You could implement `fn tags()` for all the types, and not have a trait. You're introducing more lines of code but what is the benefit?

Reply:
That's a remainder of an older version of my code. I should have removed it.